### PR TITLE
doc: extend board porting guide with new board extend feature

### PR DESCRIPTION
**Pull request by tejlmand**
Extend the board porting guide with documentation on the new board extend feature.

### CODE REVIEW

**Comment by MaureenHelm** [comment]
@kartben please take a look

**Comment by kartben** [comment]
@tejlmand  I think it would make sense if this PR would also ad a commit adding board_porting.rst somewhere in the MAINTAINERS file (maybe under the `Board/SoC configuration` section?)?

File: doc/hardware/porting/board_porting.rst [hunk 1]
```diff
@@ -922,6 +922,56 @@ There are some extra things you'll need to do:
...
+
+   boards/<VENDOR>/plank
+   ├── board.yml
+   ├── plank_<new-qualifiers>_defconfig
+   └── plank_<new-qualifiers>.dts
+
+Replace ``plank`` with the real name of the board you extend.
+
+In some cases you might want to also adjust additional settings, like the
+:file:`Kconfig.defconfig` or :file:`Kconfig.<board>`.
```
Comment (kartben) [inline doc/hardware/porting/board_porting.rst:959]:
```suggestion
:file:`Kconfig.defconfig` or :file:`Kconfig.{board}`.
```

this automagically renders as <code>Kconfig.<em>board</em><code>

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/hardware/porting/board_porting.rst [hunk 2]
```diff
@@ -922,6 +922,56 @@ There are some extra things you'll need to do:
...
+   boards/<VENDOR>/plank
+   ├── board.cmake
+   ├── Kconfig
+   ├── Kconfig.plank
+   ├── Kconfig.defconfig
+   └── plank_<new-qualifiers>.yaml
+
+Board extensions
+****************
+
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply doc/hardware/porting/board_porting.rst]: Update, ptal.

**Pull request closed without merging**